qadence_protocols/measurements/protocols.py

```diff
@@ -1,115 +1,112 @@
 from __future__ import annotations
 
 import importlib
-from dataclasses import dataclass
-from functools import partial
 
 from qadence import QuantumModel
+from qadence.blocks.abstract import AbstractBlock
 from torch import Tensor
 
+from qadence_protocols.measurements.abstract import MeasurementManager
 from qadence_protocols.protocols import Protocol
+from qadence_protocols.types import MeasurementData
 
 PROTOCOL_TO_MODULE = {
-    "tomography": "qadence_protocols.measurements.tomography",
-    "shadow": "qadence_protocols.measurements.shadow",
-    "robust_shadow": "qadence_protocols.measurements.robust_shadow",
+    "tomography": ("qadence_protocols.measurements.tomography", "Tomography"),
+    "shadow": ("qadence_protocols.measurements.shadow", "ShadowManager"),
+    "robust_shadow": ("qadence_protocols.measurements.robust_shadow", "RobustShadowManager"),
 }
 
 
-@dataclass
-class MeasurementOptions:
-    def __init__(self, protocol: str, options: dict = dict()) -> None:
-        self.protocol = protocol
-        self.options = options
+class Measurements(Protocol):
+    """Define a measurement protocol.
+
+    Possible options are available via the `MeasurementProtocols` type.
 
-        self.OPTIONS_TO_VERIFY = {
-            "shadow": self.verify_classical_shadow,
-            "robust_shadow": self.verify_robust_shadow,
-            "tomography": self.verify_tomography,
-        }
+    Attributes:
+        protocol (str): Protocol name.
+        options (dict, optional): Options to run protocol.
+    """
 
-    def verify_options(self) -> dict:
-        return self.OPTIONS_TO_VERIFY[self.protocol]()
+    def __init__(
+        self,
+        protocol: str,
+        model: QuantumModel,
+        options: dict = dict(),
+        observables: list[AbstractBlock] = list(),
+        param_values: dict[str, Tensor] = dict(),
+        state: Tensor | None = None,
+        data: MeasurementData = MeasurementData(),
```

Comment:
I guess `model`, `observables`, `param_values` and `states` can be moved to `__call__`. `data` whenever it comes to experimental data can be passed in `options` ?